glhe/aggregation/base_agg.py

```diff
@@ -1,36 +1,36 @@
-import os
 from abc import ABC, abstractmethod
+from pathlib import Path
 
 import numpy as np
 
-from glhe.utilities.functions import Interpolator1D, Interpolator1DFromFile
-from glhe.utilities.functions import Interpolator2DFromFile
-
-join = os.path.join
-norm = os.path.normpath
-cwd = os.getcwd()
+from glhe.utilities.functions import InterpolatorBase, Interpolator1D, Interpolator1DFromFile, Interpolator2DFromFile
 
 
 class BaseAgg(ABC):
 
     def __init__(self, inputs: dict):
         # g-function values
         if 'g-function-path' in inputs:
-            path_g = norm(join(cwd, inputs['g-function-path']))
-            self.interp_g = Interpolator1DFromFile(path_g)
+            p = Path(inputs['g-function-path'])
+            if not p.is_absolute():
+                p = Path.cwd() / inputs['g-function-path']
```

Comment:
This pattern is in the code pretty much anywhere we are reading a path from the input dict.  I wonder if there is a better way to do it.

And after taking a quick look, it looks like the Path's `resolve()` method will handle this.  If it's relative, it will apply `cwd()` to make it absolute.  And if it's absolute it will leave it.  And it will walk through links and stuff.  I'll fix that up.

TIL.

Reply:
Yeah, I try to `resolve()` any pathlib object at instantiation, and then it all just works. This was all developed before I was a pathlib convert. 🙏